cnd/src/swap_protocols/halight.rs

```diff
@@ -91,34 +94,34 @@ pub mod data {
 }
 
 #[derive(Default, Debug)]
-pub struct InvoiceStates {
+pub struct StateStore {
```

Comment:
We deliberately went away from this naming, did you have to introduce it again? :grimacing:

Reply:
What about naming this `States` instead and refer to it as `halight::States` everywhere?

Reply:
Best to review in chronological order with https://github.com/comit-network/comit-rs/pull/2450 first :)

Reply:
Can you please describe in more details why we went away from this naming?

Reply:
I find it to be a very awkward name because it is super generic. It also has quite an object-oriented touch which I'd consider to be an anti-pattern in Rust. Maybe @luckysori can also say something, I know he dislikes it aswell :)